library/std/src/sys/windows/thread.rs

```diff
@@ -101,11 +101,11 @@ impl Thread {
 
 pub fn available_parallelism() -> io::Result<NonZeroUsize> {
     let res = unsafe {
-        let mut sysinfo: c::SYSTEM_INFO = crate::mem::zeroed();
-        c::GetSystemInfo(&mut sysinfo);
-        sysinfo.dwNumberOfProcessors as usize
+        // FIXME: windows::Win32::System::SystemServices::ALL_PROCESSOR_GROUPS should be u16, not u32
+        c::GetActiveProcessorCount(65535u16) as usize
     };
     match res {
+        // FIXME: can call GetLastError, but what it can return?
```

Comment:
I think this can just return the `last_os_error` now. The API does not promise any specific errors are returned.